daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2128,13 +2173,13 @@ func (devices *DeviceSet) removeDevice(devname string) error {
 	return err
 }
 
-func (devices *DeviceSet) cancelDeferredRemoval(info *devInfo) error {
+func (devices *DeviceSet) cancelDeferredRemovalIfNeeded(info *devInfo) error {
 	if !devices.deferredRemove {
 		return nil
 	}
 
-	logrus.Debugf("devmapper: cancelDeferredRemoval START(%s)", info.Name())
-	defer logrus.Debugf("devmapper: cancelDeferredRemoval END(%s)", info.Name())
+	logrus.Debugf("devmapper: cancelDeferredRemovalIfNeeded START(%s)", info.Name())
+	defer logrus.Debugf("devmapper: cancelDeferredRemovalIfNeeded END(%s)", info.Name())
```

Comment:
Are you taking about these two ? 

1) https://github.com/shishir-a412ed/docker/blob/dm_task_run_failed/daemon/graphdriver/devmapper/deviceset.go#L2181-L2182

2) https://github.com/shishir-a412ed/docker/blob/dm_task_run_failed/daemon/graphdriver/devmapper/deviceset.go#L2204-L2205

`takeSnapshot()` calls `cancelDeferredRemoval()` directly.  So in that case only `cancelDeferredRemoval()` logs will be printed.